src/main/java/dev/jbang/FileRef.java

```diff
@@ -10,40 +10,34 @@
 import java.nio.file.StandardCopyOption;
 
 public class FileRef {
+	final String base;
+	final String ref;
+	final String destination;
 
-	final Script source;
-
-	String ref;
-
-	String destination;
-
-	public FileRef(Script source, String destination, String ref) {
-		assert (destination != null);
-		this.source = source;
+	public FileRef(String base, String ref, String destination) {
```

Comment:
reason I had Script passed in here was that in case of error you should be able to report specificly where it was included from at least. is base enough for that?

Reply:
The code wasn't using anything else from the Script than it's base, that's why I removed it, to simplify it as much as possible. First I changed it to (what is now) ResourceRef but then I realized it wasn't even using all of that.